File: programmers/level3/섬연결하기.py
def solution(n, costs):
    answer = 0
    costs.sort(key=lambda x: x[2])
    visited = set()
    visited.add(0)

    while len(visited) < n:
        for s, e, c in costs:
            if s in visited or e in visited:
                if s in visited and e in visited: continue
                visited.add(s)
                visited.add(e)
                answer += c
                break

    return answer


if __name__ == "__main__":
    print(solution(4, [[0,1,1],[0,2,2],[1,2,5],[1,3,1],[2,3,8]]))

File: programmers/level3/섬연결하기.kt
package test

class ConnectIsland {
    fun solution(n: Int, costs: Array<IntArray>): Int {
        val sortedCosts = costs.sortedBy { it[2] }
        val visited = mutableSetOf(0)

        var answer = 0
        while (visited.size < n) {
            for ((s, e, c) in sortedCosts) {
                if (visited.contains(s) or visited.contains(e)) {
                    if (visited.contains(s) and visited.contains(e)) {
                        continue
                    }
                    visited.add(s)
                    visited.add(e)
                    answer += c
                    break
                }
            }
        }
        return answer
    }
}

fun main() {
    val arr = arrayOf(
        intArrayOf(0,1,1),
        intArrayOf(0,2,2),
        intArrayOf(1,2,5),
        intArrayOf(1,3,1),
        intArrayOf(2,3,8)
    )
    println(ConnectIsland().solution(4, arr))
}